Update runtime to 25.08

user1-github opened the pull request:
Also remove server binary as there's no way to launch it from the Flatpak.

user1-github committed:
Update runtime to 25.08

user1-github committed:
Change to id, app-id is deprecated

user1-github commented:
bot, build

user1-github commented:
It seems to fail on manifest validation check due to

manifest-json-warnings' error found in linter manifest check. Details: ['Failed to deserialize "revision" property of type "gchararray" for an object of type "BuilderSourceSvn"']

I don't know what that means, so I'll have to ask someone what to do..

Mailaender commented on gg.tesseract.Tesseract.yaml:
```diff
@@ -28,6 +28,7 @@ modules:
       - install -Dm644 ../media/interface/cube.png ${FLATPAK_DEST}/share/icons/hicolor/256x256/apps/${FLATPAK_ID}.png
       - install -Dm644 ../gg.tesseract.Tesseract.desktop ${FLATPAK_DEST}/share/applications/${FLATPAK_ID}.desktop
       - install -Dm644 ../gg.tesseract.Tesseract.metainfo.xml ${FLATPAK_DEST}/share/metainfo/${FLATPAK_ID}.metainfo.xml
+      - rm -rf ${FLATPAK_DEST}/lib/tesseract/bin_unix/native_server ${FLATPAK_DEST}/lib/tesseract/bin_unix/readme.txt
```
You can probably still run it with `flatpak run --nodevice=dri --command=native_server gg.tesseract.Tesseract` and if that doesn't work please submit a fix rather than a removal.

user1-github replied:
Ok, I left the server binary as is. According to the documentation it should be possible to launch it via the main executable with an arg.

user1-github committed:
Leave server binary as is and fix previous manifest check failure

user1-github committed:
Add sdl2_image, it was removed in 25.08

user1-github committed:
Modify include path to make the game find the SDL2_image header file

Some games building in a Flatpak environment seem to have trouble finding SDL2 related header files if they're located in /app/include/SDL2. I had to apply the same workaround for net.redeclipse.RedEclipse-Legacy Flatpak that I maintain.

user1-github committed:
Try using a different dir for Tesseract

The game now seems to have trouble finding the SDL2_image library probably due to itself residing in /app/lib

user1-github committed:
Set ldflags in build options to hopefully make the game find SDL2_image library

user1-github committed:
Try using cmake-ninja

user1-github committed:
Try autotools again

user1-github committed:
Include paths are still needed

user1-github commented:
@Mailaender maybe you know why it always fails on 

/usr/lib/gcc/aarch64-unknown-linux-gnu/15.2.0/../../../../aarch64-unknown-linux-gnu/bin/ld: cannot find -lSDL2_image: No such file or directory ?

I've tried many options to fix this, including setting ldflags: -L/app/lib , but it still cannot detect the SDL2_image library in /app/lib.

user1-github committed:
Add files via upload

user1-github committed:
Use patch to make Tesseract find SDL2_image in /app/lib

user1-github commented:
Finally it succeeded. Seems like patching the makefile is the only way to make the linker find SDL2_image in /app/lib
